src/app/button/terra-button.component.ts

```diff
@@ -67,11 +71,34 @@ export class TerraButtonComponent
         this.inputIsHighlighted = false;
     }
 
+    ngOnInit():void
+    {
+        this._initialTooltipPlacement = this.inputTooltipPlacement;
+    }
+
     private click(event:Event):void
     {
         if(isNullOrUndefined(this.inputIsDisabled) || this.inputIsDisabled === false)
         {
             this.outputClicked.emit(event);
         }
     }
+
+    private setTooltipPlacement(event:MouseEvent):void
+    {
+        let minimalDistanceToWindowEdge:number = 100;
+
+        if(window.innerWidth - event.clientX < minimalDistanceToWindowEdge)
```

Comment:
Könnte man das ganze nicht auch gleich für den oberen und unteren Seitenrand umsetzen. `event.clientY` und `window.innerHeight` geben doch bestimmt die nötigen Informationen.

Reply:
oberer und unterer Rand ist komplizierter, wenn sich der Button in der Split-View befindet. deshalb würde ich davon absehen